Core/src/org/sleuthkit/autopsy/contentviewers/SQLiteViewer.java

```diff
@@ -363,33 +410,103 @@ protected Integer doInBackground() throws Exception {
             protected void done() {
                 super.done();
                 try {
-                    int numRows = get();
+
+                    numRows = get();
                     numEntriesField.setText(numRows + " entries");
+
+                    currPage = 1;
+                    currPageLabel.setText(Integer.toString(currPage));
+                    numPagesLabel.setText(Integer.toString((numRows / ROWS_PER_PAGE) + 1));
+
+                    prevPageButton.setEnabled(false);
+
+
+                    if (numRows > 0) {
+                        nextPageButton.setEnabled(((numRows > ROWS_PER_PAGE)));
+                        readTable(tableName, (currPage - 1) * ROWS_PER_PAGE + 1, ROWS_PER_PAGE);
+                    } else {
+                        nextPageButton.setEnabled(false);
+                        selectedTableView.setupTable(Collections.emptyList());
+                    }
+
                 } catch (InterruptedException | ExecutionException ex) {
                     LOGGER.log(Level.SEVERE, "Unexpected exception while reading table.", ex); //NON-NLS
                 }
             }
-        }.execute();
-
+        };
+        worker.execute();
     }
 
-    enum SQLStorageClass {
-        NULL,
-        INTEGER,
-        REAL,
-        TEXT,
-        BLOB
-    };
+    private void readTable(String tableName, int startRow, int numRowsToRead) {
 
-    private class SQLColDef {
+        if (worker != null && !worker.isDone()) {
+            worker.cancel(false);
+            worker = null;
+        }
 
-        private final String colName;
-        private final SQLStorageClass storageClass;
+        worker = new SwingWorker<ArrayList<Map<String, Object>>, Void>() {
+            @Override
+            protected ArrayList<Map<String, Object>> doInBackground() throws Exception {
+                try {
+                    Statement statement = connection.createStatement();
+                    ResultSet resultSet = statement.executeQuery(
+                            "SELECT * FROM " + tableName
+                            + " LIMIT " + Integer.toString(numRowsToRead)
+                            + " OFFSET " + Integer.toString(startRow - 1)
+                    ); //NON-NLS
 
-        SQLColDef(String colName, SQLStorageClass sc) {
-            this.colName = colName;
-            this.storageClass = sc;
+                    return resultSetToArrayList(resultSet);
+                } catch (SQLException ex) {
+                    LOGGER.log(Level.SEVERE, "Failed to get data for table.", ex); //NON-NLS
```

Comment:
It would be helpful for debugging purposes if we logged more information, i.e., at least the file name and the table name, and perhaps the other arguments as well. My concern is that as it is right now, we would have very little actionable information if we received a report of a problem from production or someone in the open source community.